cpp/src/arrow/chunked_array_test.cc

```diff
@@ -146,6 +146,33 @@ TEST_F(TestChunkedArray, EqualsDifferingMetadata) {
   ASSERT_TRUE(left.Equals(right));
 }
 
+TEST_F(TestChunkedArray, EqualsSameAddressWithNaNs) {
+  auto chunk0 = ArrayFromJSON(float64(), "[0, 1, 2, NaN]");
+  auto chunk1 = ArrayFromJSON(float64(), "[3, 4, 5]");
+  ASSERT_OK_AND_ASSIGN(auto result0, ChunkedArray::Make({chunk0, chunk1}, float64()));
+  // result0 has NaN values
+  ASSERT_FALSE(result0->Equals(result0));
```

Comment:
Can we use more meaningful names? For example:

```suggestion
  auto chunk_have_nan = ArrayFromJSON(float64(), "[0, 1, 2, NaN]");
  auto chunk_no_nan = ArrayFromJSON(float64(), "[3, 4, 5]");
  ASSERT_OK_AND_ASSIGN(auto chunked_array_have_nan, ChunkedArray::Make({chunk_have_nan, chunk_no_nan}, float64()));
  ASSERT_FALSE(chunked_array_have_nan->Equals(chunked_array_have_nan));
```

Reply:
Will do!